packages/core/src/state-management/repository.ts

```diff
@@ -267,10 +271,25 @@ export class Repository {
     opts: CreateOpts | UpdateOpts
   ): Promise<RunningState> {
     this.logger.verbose(`Repository apply commit to stream ${streamId.toString()}`)
-    const state$ = await this.stateManager.applyCommit(streamId, commit, opts)
-    await this.applyWriteOpts(state$, opts, OperationType.UPDATE)
-    this.logger.verbose(`Repository applied write options to stream ${streamId.toString()}`)
-    return state$
+
+    const state$ = await this._internals.load(streamId, opts)
+    this.logger.verbose(`Repository loaded state for stream ${streamId.toString()}`)
+
+    return this.executionQ.forStream(streamId).run(async () => {
+      const originalState = state$.state
+      const updatedState = await this.streamUpdater.applyCommitFromUser(originalState, commit)
+      if (StreamUtils.tipFromState(updatedState).equals(StreamUtils.tipFromState(originalState))) {
+        return state$ // nothing changed
+      }
+
+      state$.next(updatedState) // emit the new state
+
+      await this._internals._updateStateIfPinned(state$)
+      await this.applyWriteOpts(state$, opts, OperationType.UPDATE)
```

Comment:
the only functional change in this PR is moving this call to `applyWriteOpts` to be inside the execution queue.  This seems to me the easiest way to be able to decide whether or not to update the persistent streamstate in the state store based on whether or not creating the anchor request against the CAS is successful.